conformance/lang/expressions/shift-expression/left_shift.balt

```diff
@@ -0,0 +1,5522 @@
+
+function init() {
+    int? a = 256;
+    int? b = 5;
+    int? c = 0;
+    int? d = ();
+    int e = 133;
+    int f = 7;
+
+    int? x1 = a << b;
+    io:println(x1); // @output 8192
+    int? x2 = f << b;
+    io:println(x2); // @output 224
+    int? x3 = a << f;
+    io:println(x3); // @output 32768
+    int? x4 = e << b;
+    io:println(x4); // @output 4256
+
+    int? x5 = a << d;
+    io:println(x5.toBalString()); // @output ()
+    int? x6 = d << b;
+    io:println(x6.toBalString()); // @output ()
+    int? x7 = d << b;
+    io:println(x7.toBalString()); // @output ()
+
+    io:println(a << b); // @output 8192
+    io:println(a << c); // @output 256
+
+    io:println(a << f); // @output 32768
+    io:println(e << b); // @output 4256
+
+    io:println((a << d).toBalString()); // @output ()
+    io:println((d << b).toBalString()); // @output ()
+    io:println((d << d).toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test '<<' expression for optional zero and negative 'int' operands.
+Labels: int, nil-literal, optional-type, shift-expr, unary-minus, value:toBalString
+
+function init() {
+    int? a = -256;
+    int? b = -9223372036854775794;
+    int? c = 0;
+    int? d = ();
+    int e = -133;
+    int f = -9223372036854775799;
+
+    int? x1 = a << b;
+    io:println(x1); // @output -4194304
+    int? x2 = f << b;
+    io:println(x2); // @output 147456
+    int? x3 = a << f;
+    io:println(x3); // @output -131072
+    int? x4 = e << b;
+    io:println(x4); // @output -2179072
+
+    int? x5 = a << d;
+    io:println(x5.toBalString()); // @output ()
+    int? x6 = d << b;
+    io:println(x6.toBalString()); // @output ()
+
+    io:println(a << b); // @output -4194304
+    io:println(a << c); // @output -256
+
+    io:println(a << f); // @output -131072
+    io:println(e << b); // @output -2179072
+
+    io:println((a << d).toBalString()); // @output ()
+    io:println((d << b).toBalString()); // @output ()
+    io:println((d << d).toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test '<<' expression for optional positive and negative 'int' operands.
+Labels: int, optional-type, shift-expr, unary-minus
+
+function init() {
+    int? a = 256;
+    int? b = 5;
+    int? c = -256;
+    int? d = -9223372036854775799;
+    int e = 133;
+    int f = 7;
+    int g = -133;
+    int h = -9223372036854775801;
+
+    io:println(a << d); // @output 131072
+    io:println(c << b); // @output -8192
+
+    io:println(a << h); // @output 32768
+    io:println(c << f); // @output -32768
+
+    io:println(e << d); // @output 68096
+    io:println(g << b); // @output -4256
+}
+
+Test-Case: output
+Description: Test 0x3F mask of 'int' RHS operand in '<<' expression.
+Labels: int, shift-expr, unary-minus
+
+function init() {
+    int a = -1;
+    int b = 63;
+    int c = 64;
+    int d = 122;
+    int e = -65;
+    int f = -64;
+
+    io:println(a << b); // @output -9223372036854775808
+    // Explanation:
```

Comment:
Any specific reason why we decided to include an explanation of how shifting happens?
May be breaking the mask getting applied cases separately and mentioning in the description why we needed to mask the RHS?

Reply:
> Any specific reason why we decided to include an explanation of how shifting happens?
In the code review, we decided to add an explanation, because shift operator works with bit representation.

> May be breaking the mask getting applied cases separately and mentioning in the description why we needed to mask the RHS?
Spec explains about the RHS mask. So do we need to mention it here also?

Reply:
if we came into a verdict to have the explanation; then fine